src/DynamoCore/Models/AnnotationModel.cs

```diff
@@ -0,0 +1,379 @@
+                                region.Height = this.Height;
+                            }
+                            if (overlap.Rect.Left < this.Left ||
+                                    overlap.Rect.Right > region.Right) //Overlap in width - increase the region width
+                            {
+                                this.Width += overlap.Rect.Right - region.Right + 10;
+                                region.Width = this.Width;
+                            }
+                        }
+                    }
+                }
+                               
+                //Initial Height is to store the Actual height of the group.
+                this.InitialHeight = region.Height;
+            }
+        }
+
+        /// <summary>
+        /// Group the Models based on Height and Width
+        /// </summary>
+        /// <returns></returns>
+        private Tuple<Double,Double> CalculateWidthAndHeight()
+        {           
+            var xgroup = SelectedModels.OrderBy(x => x.X).ToList();
+            var ygroup = SelectedModels.OrderBy(x => x.Y).ToList();
+          
+            return Tuple.Create(xgroup.Last().Width, ygroup.Last().Height);
+        }
+        
+        /// <summary>
+        /// Deserializes the model guids from XML
+        /// and creates group on those model guids.
+        /// </summary>
+        private void DeserializeGroup()
+        {
+            var listOfModels = new List<ModelBase>();
+            foreach (var objGuid in modelGuids.Split(','))
+            {
+                Guid result;
+                if (Guid.TryParse(objGuid, out result))
+                    if (SelectedModels != null)
+                    {
+                        var model = SelectedModels.FirstOrDefault(x => x.GUID == result);
+                        listOfModels.Add(model);
+                    }
+            }
+            SelectedModels = listOfModels;           
+        }
+
+        #region Serialization/Deserialization Methods
+
+        protected override void SerializeCore(XmlElement element, SaveContext context)
+        {            
+            XmlElementHelper helper = new XmlElementHelper(element);
+            helper.SetAttribute("guid", this.GUID);
+            helper.SetAttribute("text", this.AnnotationText);
+            helper.SetAttribute("left", this.Left);
+            helper.SetAttribute("top", this.Top);
+            helper.SetAttribute("width", this.Width);
+            helper.SetAttribute("height", this.Height);
+            helper.SetAttribute("fontSize", this.FontSize);
+            helper.SetAttribute("initialTop", this.InitialTop);
+            helper.SetAttribute("initialHeight", this.InitialHeight);
+            helper.SetAttribute("annotationColor", (this.BackGroundColor == null ? "" : this.BackGroundColor.ToString()));
+            helper.SetAttribute("ModelGUIDs", string.Join(",", this.SelectedModels.Select(x => x.GUID)));           
```

Comment:
Deserializing this element will emit a whole bunch of `PropertyChanged` events.  Why not set the private backing fields?